### DISCUSSION TITLE
Extending Zigbee2mqtt to outbuilding 30m/100ft away

### DISCUSSION THREAD

**Discussion by 877dev** [Koenkk/zigbee2mqtt#17129] [General]
I’d like to extend my zigbee2mqtt network to a garden building 30m/100ft away from my house. I have ethernet wiring there too.  I see three options:

1) Use ZZH stick at each end - assume would be ok at this distance?
2) Add a Raspberry Pi running a second instance of Zigbee2mqtt connected to LAN 
3) Some hard wired option - is it even possible though. 

What’s everyone’s thoughts/experience?

**Comment by geniesek** [comment]
I would use second (slave) coordinator that will via TCP/IP communicate with your primary coordinator. 
In that case your control unit can listen to messages published from primary coordinator (that will contain also secondary coordinator messages)

**Reply by andreapx** [reply]
> I would use second (slave) coordinator that will via TCP/IP communicate with your primary coordinator. In that case your control unit can listen to messages published from primary coordinator (that will contain also secondary coordinator messages)

How do the 2 coordinator communicate with each other?

**Reply by lux73** [reply]
AFAIK this is not possible ^^

**Comment by junosuarez** [comment]
I don't have experience with doing this, so others may have different thoughts, but here's mine: do you need it to be a single Zigbee mesh, or would it work to have two separate Zigbee networks joined by ethernet? If you don't have devices physically moving between both locations, and if you don't need to use zigbee-level binding, then it might be simpler and more reliable to have two separate networks with a controller in each location, with zigbee2mqtt configured to use a separate mqtt base topic for each network. They can use the same mqtt broker, and you'd be able to monitor all of the devices across both locations.

As I understand, Zigbee can only have a single coordinator per network. So if you want to get a single zigbee network to extend out to that location, you'd need to chain router (repeater) devices along the way to make the signal reach.

**Comment by sergge1** [comment]
@877dev 
> I’d like to extend my zigbee2mqtt network to a garden building 30m/100ft away from my house. I have ethernet wiring there too. I see three options:
> What’s everyone’s thoughts/experience?

Just use additional any Zigbee-Ethernet adapter, and that is it - [TubeZB](https://tubeszb.com/product/cc2652p2-based-zigbee-to-poe-coordinator-2022/) or [SMLIGHT SLZB-06](https://smlight.tech/product/slzb-06)


Also, have a look at this article on Reddit https://www.reddit.com/r/homeassistant/comments/z28f9b/my_experience_with_multiple_zigbee_poe/- the user solved exactly the same problem as you need to solve.

**Reply by pspeirs** [reply]
Not much in the way of detailing how he actually achieved it though.

**Reply by pspeirs** [reply]
So to follow on, I ran a couple of experiments here and found that it is possible and quite easy to set up.   You would end up with two co-ordinators talking to the MQTT broker.

Set up an instance of Zigbee2MQTT Edge on a Proxmox LCX, assuming you have this running.  Connect a Zigbee gateway adapter POE to the network and configure Z2M to use it.  You could potentially have as many individual LCX containers as needed.  Just rename the topic so there is no clash with the system messages.

**Reply by lux73** [reply]
for sure two z2m Instances with two Coordinator with different mqtt Topic is possible 😎
